scripts/core/helpers/network.tsx

```diff
@@ -173,3 +174,30 @@ export function uploadFileWithProgress<T>(
             });
         });
 }
+
+export function trackArticleActionProgress<T>(
+    p: () => Promise<T>,
```

Comment:
`getPromise` would be a better name. `p` was a placeholder in my initial snippet.